snippets/Alert.md

```diff
@@ -0,0 +1,107 @@
+---
+title: Alert
+tags: visual,beginner,state,effect
+---
+
+Creates an alert component with `type` prop.
+
+- Define appropriate CSS styles and animations for the component's elements.
+- Use the `React.setState()` hook to create the `isShown` and `isLeaving` state variables and set their values to `false`.
+- Define `timeoutId` to keep the timer instance for clearing on component unmount.
+- Use the `React.setEffect()` hook to update the value of `isShown` to `true` and clear interval by using `timeoutId` when component is unmounted.
+- Define `closeNotification` function to set the component is removed from DOM by displaying fading out animation and set `isShown` to `false` via `setTimeout()`. 
+- Define the component, which renders alert component with user defined `message` and a close button to remove the component from DOM.
+
+```css
+@keyframes leave {
+  0% { opacity: 1 }
+  100% { opacity: 0 }
+}
+
+.alert {
+  padding: 0.75rem 0.5rem;
+  margin-bottom: 0.5rem;
+  text-align: left;
+  padding-right: 40px;
+  border-radius: 4px;
+  font-size: 16px;
+  position: relative;
+}
+
+.alert.success {
+  color: #155724;
+  background-color: #d4edda;
+  border-color: #c3e6cb;
+}
+
+.alert.info{
+  color: #004085;
+  background-color: #cce5ff;
+  border-color: #b8daff;
+}
+
+.alert.warning{
+  color: #856404;
+  background-color: #fff3cd;
+  border-color: #ffeeba;
+}
+
+.alert.danger{
+  color: #721c24;
+  background-color: #f8d7da;
+  border-color: #f5c6cb;
+}
+
```

Comment:
A few too many styles here for my taste. Just keep the `.warning` and `.error` types, should be enough to get the general idea and it's easy to extend to more types by the user.

Reply:
Yep, removed `.info` and `.success`.